### PR TITLE
added referred to facility field in discharge form

### LINKED ISSUE #5407: Referred to facility- discharge form
**Is your feature request related to a problem? Please describe.**
Currently, in discharge form, we capture when discharge is done for referral. 

But we dont capture to which hospital referral has happened. 

**Describe the solution you'd like**
https://care.coronasafe.in/facility/6dca552f-aba1-4ca5-afc1-a71744fc3d29/patient/4cdb4510-bfe5-4f41-b1d9-c0fdd27f77ad/consultation/f0dd3d31-b6db-4cc7-bc96-daa207e61a08

Add a field in the discharge pop-up right below "reason for discharge" when it is marked as referral,
Add field for referred to hospital. The field must be free text with suggest of hospital appearing as dropdown from CARE Network. However, the user must be able to input name of any hospital outside of CARE as well.

It must be a non-mandatory field, since sometimes, the doctors might simply ask patient to go to any highier center without specifying any one hospital. In those cases, he field may be left empty

### PR DESCRIPTION
**Pull request by sachdevavaibhav**
### WHAT
Added referred to facility field in discharge form

## Proposed Changes

- Fixes #5407 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### HOW
- Fetches list of facilities from backend
- created facilities local state 
- used autocomplete component

### CODE REVIEW

**Comment by sachdevavaibhav** [comment]
@rithviknishad Kindly review it. The field can search from the options but it is not able to add values which are not present. Please suggest how can I solve it?

**Comment by sachdevavaibhav** [comment]
@rithviknishad @nihal467

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Use `FacilitySelect` component instead of manually duplicating the facility fetching code.

**Comment by sachdevavaibhav** [comment]
@rithviknishad Kindly review the changes.

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Are you sure this works without any backend changes required?

**Comment by sachdevavaibhav** [comment]
@rithviknishad 
Upon my analysis I found the value is not being stored in the database. Its storing a null value for it.
I found this by setting up care backend locally. It is storing other fields like discharge_reason etc correctly but storing null for referred_to_external.

**Kindly see the last sql query in the attached screenshot.**

image

**Update:**
@rithviknishad 
I found in the backend that the data is being received properly but `PatientConsultationDischargeSerializer` is not having `referred_to_external ` field defined. Upon adding that field everything works fine. 

image

**Comment by sachdevavaibhav** [comment]
@rithviknishad 
Should I create issue and link a PR for the backend?

**Comment by rithviknishad** [comment]
@sachdevavaibhav yes! Go ahead!

**Comment by sachdevavaibhav** [comment]
> @sachdevavaibhav yes! Go ahead!

Done! Kindly review it.

**Review by rithviknishad — APPROVED** [review]

**Comment by sachdevavaibhav** [comment]
@rithviknishad Changes are merged in the backend. You may merge this as well.

**Review by rithviknishad — APPROVED** [review]

File: src/Components/Facility/DischargeModal.tsx [hunk 1]
```diff
@@ -169,6 +174,15 @@ const DischargeModal = ({
 
   const prescriptionActions = PrescriptionActions(consultationData.id ?? "");
 
+  const handleFacilitySelect = (selected: FacilityModel) => {
+    setFacility(selected ? selected : facility);
+    const { name } = selected;
+    setPreDischargeForm((prev) => ({
+      ...prev,
+      referred_to_external: name || "",
```
Comment (Ashesh3) [inline src/Components/Facility/DischargeModal.tsx]: The purpose of `referred_to_external` is to store facilities which are not present in our system. In case the facility does exist, we are to store it in `referred_to` and leave `referred_to_external` blank. You are storing the selected facility unconditionally as `referred_to_external` without checking if it is indeed external facility or not. 

Look at the code implementaion here for reference on how to do the above check: 
https://github.com/coronasafe/care_fe/blob/develop/src/Components/Facility/ConsultationForm.tsx#L744-L758

In summary:
- User selects a facility which is in our system -> send it as `referred_to: "facility-id-here"`
- User types and selects a facility which is not in our system -> send it as `referred_to_external: "name of facility"`

**Review by Ashesh3 — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@sachdevavaibhav what is the status on this PR

**Comment by sachdevavaibhav** [comment]
> @sachdevavaibhav what is the status on this PR

solved merge conflict. pushed necessary changes. Kindly review it.

@Ashesh3

File: src/Components/Facility/DischargeModal.tsx [hunk 2]
```diff
@@ -169,6 +175,17 @@ const DischargeModal = ({
 
   const prescriptionActions = PrescriptionActions(consultationData.id ?? "");
 
+  const handleFacilitySelect = (selected: FacilityModel) => {
+    setFacility(selected ? selected : facility);
+    const { id, name } = selected;
+    const isExternal = id === -1;
+    setPreDischargeForm((prev) => ({
+      ...prev,
+      referred_to: isExternal ? null : name,
```
Comment (Ashesh3) [inline src/Components/Facility/DischargeModal.tsx]: We need to send 

- The facility ID in case of `referred_to` (you're sending the name here) 

- The facility name in case of `referred_to_external` (this part is done correctly).

**Review by Ashesh3 — CHANGES_REQUESTED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by Ashesh3** [comment]
Waiting for backend PR: https://github.com/coronasafe/care/pull/1506

**Comment by nihal467** [comment]
@sachdevavaibhav can you fix the merge conflict and get it ready for testing, we need to get it merge ASAP

**Comment by sachdevavaibhav** [comment]
> @sachdevavaibhav can you fix the merge conflict and get it ready for testing, we need to get it merge ASAP

solved merge conflict. Though I have one question why are we using moment.js and day.js both. Don't they serve the same purpose. I have kept the code coming from develop which is using dayjs.

@nihal467 @Ashesh3

File: src/Components/Facility/DischargeModal.tsx [hunk 3]
```diff
@@ -130,6 +137,18 @@ const DischargeModal = ({
       return;
     }
 
+    const dischargeDetails = {
+      ...preDischargeForm,
+      discharge: value,
+      discharge_date: moment(preDischargeForm.discharge_date).toISOString(true),
```
Comment (rithviknishad) [inline src/Components/Facility/DischargeModal.tsx]: @sachdevavaibhav

- [ ] Merge the latest develop (because moment has been uninstalled)
- [ ] Discard the `package-lock.json` diff
- [ ] replace `moment` with `dayjs`

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/DischargeModal.tsx (hunk 3 above)
Comment (sachdevavaibhav) [reply src/Components/Facility/DischargeModal.tsx]: @rithviknishad Kindly review

**Comment by nihal467** [comment]
LGTM
